dspace-api/src/main/java/org/dspace/app/profile/ResearcherProfileServiceImpl.java

```diff
@@ -0,0 +1,352 @@
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private WorkspaceItemService workspaceItemService;
+
+    @Autowired
+    private InstallItemService installItemService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Autowired
+    private CollectionService collectionService;
+
+    @Autowired
+    private SearchService searchService;
+
+    @Autowired
+    private GroupService groupService;
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @Override
+    public ResearcherProfile findById(Context context, UUID id) throws SQLException, AuthorizeException {
+        Assert.notNull(id, "An id must be provided to find a researcher profile");
+
+        Item profileItem = findResearcherProfileItemById(context, id);
+        if (profileItem == null) {
+            return null;
+        }
+
+        return new ResearcherProfile(profileItem);
+    }
+
+    @Override
+    public ResearcherProfile createAndReturn(Context context, EPerson ePerson)
+            throws AuthorizeException, SQLException, SearchServiceException {
+
+        Item profileItem = findResearcherProfileItemById(context, ePerson.getID());
+        if (profileItem != null) {
+            throw new ResourceAlreadyExistsException("A profile is already linked to the provided User");
+        }
+
+        Collection collection = findProfileCollection(context)
+            .orElseThrow(() -> new IllegalStateException("No collection found for researcher profiles"));
+
+        context.turnOffAuthorisationSystem();
+        try {
+
+            Item item = createProfileItem(context, ePerson, collection);
+            return new ResearcherProfile(item);
+
+        } finally {
+            context.restoreAuthSystemState();
+        }
+
+    }
+
+    @Override
+    public void deleteById(Context context, UUID id) throws SQLException, AuthorizeException {
+        Assert.notNull(id, "An id must be provided to find a researcher profile");
+
+        Item profileItem = findResearcherProfileItemById(context, id);
+        if (profileItem == null) {
+            return;
+        }
+
+        if (isHardDeleteEnabled()) {
+            deleteItem(context, profileItem);
+        } else {
+            removeOwnerMetadata(context, profileItem);
+        }
+
+    }
+
+    @Override
+    public void changeVisibility(Context context, ResearcherProfile profile, boolean visible)
+            throws AuthorizeException, SQLException {
+
+        if (profile.isVisible() == visible) {
+            return;
+        }
+
+        Item item = profile.getItem();
+        Group anonymous = groupService.findByName(context, ANONYMOUS);
+
+        if (visible) {
+            authorizeService.addPolicy(context, item, READ, anonymous);
+        } else {
+            authorizeService.removeGroupPolicies(context, item, anonymous);
+        }
+
+    }
+
+    @Override
+    public ResearcherProfile claim(Context context, EPerson ePerson, URI uri)
```

Comment:
@LucaGiamminonni : Thanks for updating the CanClaimItemFeature to check for a matching email address.  However, it looks like you forgot to make similar updates to the `claim()` code itself.  So, if you use the REST API directly, it looks like any user can still claim any Profile.

Could you update the REST API claim code, and add ITs to verify that it's not possible for a normal user to claim a profile which doesn't match their email address?

Thanks!